qt-models/filtermodels.cpp

```diff
@@ -32,30 +32,38 @@ FilterModelBase::FilterModelBase(QObject *parent) : QStringListModel(parent),
 {
 }
 
-// Update the stringList and the checkState array.
+// Update the stringList and the items array.
 // The last item is supposed to be the "Show Empty Tags" entry.
 void FilterModelBase::updateList(const QStringList &newList)
 {
-	// Keep copy of old checkState array to reimport them later.
-	std::vector<char> oldCheckState = checkState;
-	checkState.resize(newList.count());
-	std::fill(checkState.begin(), checkState.end(), false);
+	// Keep copy of the old items array to reimport the checked state later.
+	// Note that by using std::move(), this is an essentially free operation:
+	// The data is moved from the old array to the new one and the old array
+	// is reset.
+	std::vector<Item> oldItems = std::move(items);
+	items.resize(newList.count());
+
+	// First, reset all checked states to false
 	anyChecked = false;
+	for (Item &item: items)
```

Comment:
inconsistent whitespace - this is one construct that I don't think we list in CodingStype. Here you use 
`for (type single: vector)` with no space before the `:`, below in line 124 there's a space before the `:`

Reply:
Added a white space commit. The version without space seems to be the more common one.